src/test/java/com/xceptance/xlt/engine/htmlunit/okhttp3/CookieJarImplTest.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2005-2025 Xceptance Software Technologies GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.xceptance.xlt.engine.htmlunit.okhttp3;
+
+import org.htmlunit.util.Cookie;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+import util.JUnitParamsUtils;
+import util.JUnitParamsUtils.BlankStringOrNullParamProvider;
+import util.JUnitParamsUtils.BlankStringParamProvider;
+
+@RunWith(JUnitParamsRunner.class)
+public class CookieJarImplTest
+{
+    @Test
+    @Parameters(source = BlankStringParamProvider.class)
+    public void toOkHttpCookie_BlankDomain(final String domain)
+    {
+        callToOkHttpCookieButExpectException(new Cookie(domain, "name", "value", "/", null, false, false), CookieJarImpl.ERROR_MSG_DOMAIN);
+    }
+
+    @Test
+    @Parameters(source = DomainDataProvider.class)
+    public void toOkHttpCookie_ValidDomain(final String domain, final boolean expectedHostOnly)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie(domain, "name", "value", "/", null, false, false));
+
+        Assert.assertEquals("example.com", okCookie.domain());
+        Assert.assertEquals(expectedHostOnly, okCookie.hostOnly());
+    }
+
+    @Test
+    @Parameters(source = BlankStringParamProvider.class)
+    public void toOkHttpCookie_BlankName(final String name)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", name, "value", "/", null, false, false));
+
+        Assert.assertEquals("", okCookie.name());
+    }
+
+    @Test
+    @Parameters(source = NameProvider.class)
+    public void toOkHttpCookie_ValidName(final String name)
+    {
+        final var okCookie = CookieJarImpl.toOkHttpCookie(new Cookie("domain", name, "value", "/", null, false, false));
+
+        Assert.assertEquals("name", okCookie.name());
```

Comment:
This works, but maybe we should change the validation to this:

```suggestion
        Assert.assertEquals(name.trim(), okCookie.name());
```

So the test would still succeed if we ever change or expand the test data set.
(The same goes for the ValidValue and ValidPath tests)